frontend/src/components/organisms/ManageUsers.tsx

```diff
@@ -147,76 +163,154 @@ const Active = ({
 
 /** A ManageUsers component */
 const ManageUsers = ({}: ManageUsersProps) => {
+  const queryClient = useQueryClient();
+
   /** Pagination model for the table */
   const [paginationModel, setPaginationModel] = useState<GridPaginationModel>({
     page: 0,
-    pageSize: 20,
+    pageSize: 10,
   });
-  let cursor = "";
 
-  /** If a valid cursor is passed, fetch the next batch of users */
-  const fetchUsersBatch = async (cursor?: string) => {
-    if (cursor !== "") {
+  /** Sorting Model for the table */
+  const [sortModel, setSortModel] = React.useState<GridSortModel>([
+    { field: "firstName", sort: "asc" },
+  ]);
+
+  /** Function to fetch a batch of users with respect to the current pagination
+   * and sorting states.
+   * @param cursor is the cursor to fetch the next batch of users
+   * @param prev is a boolean to determine if the function is being called to fetch the previous page
+   * Note: If the prev boolean is true, we pass a negative value to the limit parameter to fetch
+   * data before the current cursor. This is effectively the previous page.
+   * @returns the current page of users (with respect to the current pagination and sorting states)
```

Comment:
I tried to put comments on some of these functions so they are not too difficult to follow, the order I put them is a good way to understand what they are doing